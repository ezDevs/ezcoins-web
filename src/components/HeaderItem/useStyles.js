import { makeStyles } from '@material-ui/core';
import { fade } from '@material-ui/core/styles';

export default makeStyles(theme => ({
  grow: {
    flexGrow: 1,
  },
  imageMarked: {
    height: 1,
    width: 20,
    backgroundColor: theme.palette.common.white,
    position: 'absolute',
    bottom: 12,
    left: 'calc(50% - 10px)',
  },
  profile: {
    height: 40,
    width: 50,
  },
  profileContainer: {
    alignContent: 'center',
    backgroundColor: fade(theme.palette.primary.light, 0.5),
    height: 55,
    marginLeft: 20,
  },
  sectionDesktop: {
    display: 'none',
    [theme.breakpoints.up('md')]: {
      display: 'flex',
    },
  },
  sectionMobile: {
    display: 'flex',
    [theme.breakpoints.up('md')]: {
      display: 'none',
    },
  },
  title: {
    display: 'none',
    [theme.breakpoints.up('sm')]: {
      display: 'block',
    },
  },
}));
