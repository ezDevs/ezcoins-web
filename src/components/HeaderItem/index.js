import React, { memo, useCallback } from 'react';
import { Button } from '@material-ui/core';
import { useHistory, useLocation } from 'react-router-dom';
import clsx from 'clsx';
import PropTypes from 'prop-types';

import useStyles from './useStyles';

const HeaderItem = memo(({ path, title }) => {
  const classes = useStyles();

  const history = useHistory();

  const { pathname } = useLocation();

  const onClick = useCallback(() => history.push(path), [history, path]);

  return (
    <Button color="inherit" onClick={onClick}>
      {title}
      <span className={clsx(pathname === path && classes.imageMarked)} />
    </Button>
  );
});

HeaderItem.propTypes = {
  path: PropTypes.string,
  title: PropTypes.string,
};

HeaderItem.defaultProps = {
  path: 'path',
  title: '',
};

export default HeaderItem;
