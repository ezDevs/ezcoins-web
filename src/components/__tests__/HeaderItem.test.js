import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';

import HeaderItem from '../HeaderItem';

describe('testing header item component', () => {
  it('should renders without props as expected', () => {
    const { container } = render(
      <MemoryRouter>
        <HeaderItem />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
