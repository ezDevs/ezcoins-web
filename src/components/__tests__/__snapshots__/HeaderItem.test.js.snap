// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing header item component should renders with props as expected 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Test
      <span
        class=""
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`testing header item component should renders without props as expected 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      
      <span
        class=""
      />
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
